tools/commands-packages.js

```diff
@@ -276,6 +289,63 @@ main.registerCommand({
     return 1;
   }
 
+  // If we just want to update the package metadata, then we have all we
+  // need. Don't bother building the package.
+  if (options.update) {
+    var name = packageSource.name;
+    var version = packageSource.version;
+
+    // You can't change the metadata of a record that doesn't exist.
+    var existingRecord =
+          catalog.official.getVersion(packageSource.name, packageSource.version);
+    if (! existingRecord) {
+      Console.error(
+        "You can't call",  Console.command("`meteor publish --update`"),
+        "on version " + version + " of " + "package '" + name +
+          "' without publishing it first.");
+      return 1;
+    }
+
+    // Load in the user's documentation, and check that it isn't blank.
+    var readmeInfo;
+    main.captureAndExit(
+      "=> Errors while publishing:", "reading documentation",
+      function () {
+       readmeInfo = packageSource.processReadme();
+    });
+
+    // You are still not allowed to upload a blank README.md.
+    if (readmeInfo && readmeInfo.hash === files.blankHash) {
+      Console.error(
+        "Please fill it out your documentation file! If you don't want to",
+        "update new documentation, you can remove it out from the package",
```

Comment:
Yup, sorry, this isn't ready for a second code review yet! I am going to think about how this should work in a bit more detail, since you are finding it confusing.